theano/tests/test_rop.py

```diff
@@ -259,6 +259,33 @@ def test_rebroadcast(self):
             self.x[:4].dimshuffle('x', 0), 0).sum(axis=1),
             (1,))
 
+    def test_downsample(self):
+        rng = numpy.random.RandomState(utt.fetch_seed())
+        maxpoolshps = ((1,1),(3,2),(2,3))
+        vx = (rng.rand(2,3,3,4) * 2.0).astype(theano.config.floatX)
+        vv = (rng.rand(2,3,3,4) * 2.0).astype(theano.config.floatX)
+        input = theano.shared(vx)
+        eval_p = theano.shared(vv)
+        for maxpoolshp in maxpoolshps:
+            for ignore_border in [False, True]:
```

Comment:
If you add the padding/strides support, then you need to add extra loop to tests those condition.

Also, I think that now we have a better automatic test that is documented here:

http://www.deeplearning.net/software/theano/tutorial/extending_theano.html#testing-the-rop